api/envoy/api/v2/route/route.proto

```diff
@@ -773,6 +773,22 @@ message RouteAction {
     google.protobuf.BoolValue enabled = 2;
   };
   repeated UpgradeConfig upgrade_configs = 25;
+
+  // Configures :ref:`internal redirect <arch_overview_internal_redirects>` behavior.
+  enum RedirectAction {
+    PASS_THROUGH_REDIRECT = 0;
+    HANDLE_REDIRECT = 1;
+  }
+  // TODO(alyssawilk) it's really annoying to have a completely seprate enum
+  // because proto3 doesn't have defaults. Check with htuch@ for better
```

Comment:
I think maybe you could use an EnumValue (?) WKT and then have different defaults?